IndoorMapsBackend/src/resolvers/BuildingResolver.ts

```diff
@@ -0,0 +1,117 @@
+import 'reflect-metadata'
+import { Resolver, Query, Mutation, Arg, Ctx, InputType, Field, FieldResolver, Root, Float} from 'type-graphql'
+import { GraphQLError } from 'graphql'
+import { Floor as DbFloor } from '@prisma/client'
+
+import { Context } from '../utils/context.js'
+import { Building } from '../graphqlSchemaTypes/Building.js'
+import { convertToGraphQLBuilding, convertToGraphQLFloor } from '../utils/typeConversions.js'
+import { getUserOrThrowError } from '../auth/validateUser.js'
+import { Floor } from '../graphqlSchemaTypes/Floor.js'
+
+@InputType()
+class BuildingUniqueInput {
+    @Field()
+    id: number
+}
+
+@InputType()
+class BuildingCreateInput {
+    @Field()
+    title: string
+
+    @Field()
+    address: string
+
+    @Field(type => Float)
+    startLat: number
+
+    @Field(type => Float)
+    startLon: number
+}
+
+@Resolver(of => Building)
+export class BuildingResolver {
+    @FieldResolver((type) => [Floor]!)
+    async floors(
+        @Root() building: Building,
+        @Ctx() ctx: Context,
+    ) {
+        //TODO: investigate why using findUnique throws a error
+        const dbFloors = await ctx.prisma.building.findFirst({
```

Comment:
what does dbFloor denote? Database Floor?

Reply:
yes, The database and the graphql do not share types so this is called dbFloors to indicate that it is of the DB type (it is later converted to the Graphql type)